internals/search.py

```diff
@@ -175,21 +175,31 @@ def _resolve_promise_to_id_list(promise):
     return id_list
 
 
-def process_query(user_query, show_unlisted=False):
+def process_query(
+    user_query, sort_spec=None, show_unlisted=False, show_deleted=False,
+    start=0, num=100):
```

Comment:
Optional: instead of `start=0, num=100`, we can just do `page=0` and fix pagination to 100 items, unless we want to give user the ability to decide how many items in a page

Reply:
I'd like to keep the ability to change the page size, even if we don't expose it to users at first.